### ISSUE TITLE
About replicating your results in Table 1 for NTGA noise

### ISSUE THREAD

**Issue by saigontrade88** [fshp971/robust-unlearnable-examples#4]
Hello, 

The result for adversarial training is so good. You listed several models in your study. I am trying to replicate the results in Table 1 for NTGA noise. I used the data generated by the NTGA noise's authors. Which model did you use to generate the results for Table 1? 

I am having a low test and robust accuracy using this setting shown below. My current result is shown in the screenshot. ![vgg16-AT-NTGA-low-performance](https://github.com/fshp971/robust-unlearnable-examples/assets/35924987/8de9ca89-50db-4a81-9401-19d3d2915edd)

I loaded the vgg16 and vgg19 from torchvison without any modification. Does it cause the low performance issue? 

If it is possible, can you give some guidance on how to import other data protection datasets in your code base such as the NTGA noise? Many thanks. 

My setting is as follows.
```
python train_NTGA.py \
    --data-mode manual \
    --arch vgg16 \
    --dataset cifar10 \
    --train-steps 40000 \
    --batch-size 128 \
    --optim sgd \
    --lr 0.1 \
    --lr-decay-rate 0.1 \
    --lr-decay-freq 16000 \
    --weight-decay 5e-4 \
    --momentum 0.9 \
    --pgd-radius 4 \
    --pgd-steps 10 \
    --pgd-step-size 0.08 \ #pgd-radius/5
    --pgd-random-start \
    --pgd-norm-type l-infty \
    --report-freq 1000 \
    --data-dir ./data \
    --man_x_train_path ./data/ntga/x_train_cifar10_ntga_cnn_best.npy \
    --man_y_train_path ./data/ntga/y_train_cifar10.npy \
    --save-dir ./exp_data/cifar10/train/NTGA/exp2 \
    --save-name AT_NTGA_noise
```

I use the following class to convert the numpy arrays to Pillow Image objects. The class inherits the Pytorch Dataset class. The processed features are in the range [0, 255]. Can you help check if this class is compatible with your code? Many thanks.

```
class Cifar10NTGADataset(Dataset):
    def __init__(self, data_X, data_y, transform_list=None, fitr=None):
        '''
        data_X: numpy array, shape (40000, 32, 32, 3).
        data_Y: numpy array, shape (40000, 10)
        '''
        self.data = data_X
        self.targets = np.argmax(data_y, axis=1).tolist() # Return a Python list
              
        self.transforms = transform_list

    def __getitem__(self, index):
        img, target = self.data[index], self.targets[index]
        # Check the data type of the array
        # Add 0.5 after unnormalizing to [0, 255] to round to nearest integer
        img = np.clip(img * 255 + 0.5, 0, 255).astype(np.uint8)
       
        # doing this so that it is consistent with all other datasets
        # to return a PIL Image                                
        img = Image.fromarray(img)

        if self.transforms:
          #for transform in self.transforms: 
            img = self.transforms(img)
        return img, target
    def __len__(self):
        return len(self.data) 
```

**Comment by fshp971** [comment]
Hi,

Your `--pgd-step-size` should be `0.8` rather than `0.08`.

**Comment by saigontrade88** [comment]
Your suggestion is true. Many thanks.
